extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/AbstractPathMatchingHttpSecurityPolicy.java

```diff
@@ -151,30 +153,35 @@ private static String getAuthMechanismName(RoutingContext routingContext,
 
     private static void addPermissionToPathMatcher(Map<String, HttpSecurityPolicy> permissionCheckers, String rootPath,
             PolicyMappingConfig policyMappingConfig,
-            ImmutablePathMatcher.ImmutablePathMatcherBuilder<List<HttpMatcher>> builder) {
+            ImmutablePathMatcher.ImmutablePathMatcherBuilder<Set<HttpMatcher>> builder) {
         HttpSecurityPolicy checker = permissionCheckers.get(policyMappingConfig.policy);
         if (checker == null) {
             throw new RuntimeException("Unable to find HTTP security policy " + policyMappingConfig.policy);
         }
 
         if (policyMappingConfig.enabled.orElse(Boolean.TRUE)) {
-            for (String path : policyMappingConfig.paths.orElse(Collections.emptyList())) {
-                path = path.trim();
-                if (!path.startsWith("/")) {
-                    path = rootPath + path;
+            policyMappingConfig.paths.ifPresent(new Consumer<List<String>>() {
```

Comment:
btw changes between lines 163 and 176 can be almost completely discarded. In my first plan I needed to pass paths down so I changed it to this. I'll drop it after your review as it is not relevant.